test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -289,70 +288,176 @@ public void PlattCalibratorOnnxConversionTest()
 
             var initialPipeline = mlContext.Transforms.ReplaceMissingValues("Features").
                 Append(mlContext.Transforms.NormalizeMinMax("Features"));
+            return (mlContext, dataView, estimators, initialPipeline);
+        }
+
+        [Fact]
+        public void PlattCalibratorOnnxConversionTest()
+        {
+            // Step 1: Test calibrator with binary prediction trainer
+            var (mlContext, dataView, estimators, initialPipeline) = GetEstimatorsForOnnxConversionTests();
             foreach (var estimator in estimators)
             {
                 var pipeline = initialPipeline.Append(estimator).Append(mlContext.BinaryClassification.Calibrators.Platt());
                 var onnxFileName = $"{estimator}-WithPlattCalibrator.onnx";
 
                 TestPipeline(pipeline, dataView, onnxFileName, new ColumnComparison[] { new ColumnComparison("Score", 3), new ColumnComparison("PredictedLabel"), new ColumnComparison("Probability", 3) });
             }
-            Done();
-        }
 
-        class PlattModelInput
-        {
-            public bool Label { get; set; }
-            public float Score { get; set; }
-        }
+            // Step 2: Test calibrator without any binary prediction trainer
```

Comment:
I have consolidated the separate unit tests of calibrators with/without standard date and binary prediction trainers into 1 unit test per calibrator. If there is any better way of doing this (such as possibly passing `Platt()`, `Platt(slope, offset)`, `Isotonic()` and `Naive()` through functions), please let me know.

Reply:
Yes, that is the right approach. Is there a problem passing in the calibrator as a parameter to a function that consolidates all the steps?

---
In reply to: [451384234](https://github.com/dotnet/machinelearning/pull/5289#discussion_r451384234) [](ancestors = 451384234)